packages/jellyfish-wallet-mnemonic/src/mnemonic.ts

```diff
@@ -151,26 +169,34 @@ export class MnemonicHdNode implements WalletHdNode {
   }
 }
 
+export interface SigningInterface {
+  beforeSign: (buffer: Buffer, tx: Transaction) => Promise<void>
+  afterSign: (buffer: Buffer, tx: TransactionSegWit) => Promise<void>
+}
+
 /**
  * Provider that derive MnemonicHdNode from root. Uses a lite on demand derivation.
  */
 export class MnemonicHdNodeProvider implements WalletHdNodeProvider<MnemonicHdNode> {
+  signingCb?: SigningInterface
+
   /**
    * @param {Buffer} seed of the hd node
    * @param {Bip32Options} options for chain agnostic generation of public/private keys
    */
-  static fromSeed (seed: Buffer, options: Bip32Options): MnemonicHdNodeProvider {
+  static fromSeed (seed: Buffer, options: Bip32Options, signingCb?: SigningInterface): MnemonicHdNodeProvider {
     const node = bip32.fromSeed(seed, options)
-    return new MnemonicHdNodeProvider(node)
+    return new MnemonicHdNodeProvider(node, signingCb)
   }
 
   private readonly root: bip32.BIP32Interface
 
-  private constructor (root: bip32.BIP32Interface) {
+  private constructor (root: bip32.BIP32Interface, signingCb?: SigningInterface) {
     this.root = root
+    this.signingCb = signingCb
   }
```

Comment:
```suggestion
  private constructor (
    private readonly root: bip32.BIP32Interface, 
    private readonly signingCb?: SigningInterface) {
  }
```

Recently found out you could do this in TS.